hashing/src/lib.rs

```diff
@@ -586,4 +620,89 @@ mod tests {
 
         assert_eq!(expected, serialized);
     }
+
+    #[test]
+    fn merkle_roots_are_preimage_resistent() {
+        // Input data is two chunks long.
+        //
+        // The resulting tree will look like this:
+        //
+        // 1..0  a..j
+        // │     │
+        // └─────── R
+        //
+        // The merkle root is thus: R = h( h(1..0) || h(a..j) )
+        //
+        // h(1..0) = 807f1ba73147c3a96c2d63b38dd5a5f514f66290a1436bb9821e9f2a72eff263
+        // h(a..j) = 499e1cdb476523fedafc9d9db31125e2744f271578ea95b16ab4bd1905f05fea
+        // R=h(h(1..0)||h(a..j)) = 1319394a98d0cb194f960e3748baeb2045a9ec28aa51e0d42011be43f4a91f5f
+        // h(2u64le || R) = c31f0bb6ef569354d1a26c3a51f1ad4b6d87cef7f73a290ab6be8db6a9c7d4ee
+        //
+        // The final step is to hash h(2u64le || R), which is the length as little endian
+        // concatenated with the root.
+
+        // Constants used here assume a chunk size of 10 bytes.
+        assert_eq!(ChunkWithProof::CHUNK_SIZE_BYTES, 10);
+
+        let long_data = b"1234567890abcdefghij";
+        assert_eq!(long_data.len(), ChunkWithProof::CHUNK_SIZE_BYTES * 2);
+
+        // The `long_data_hash` is constructed manually here, as `Digest::hash` still had
+        // deactivated chunking code at the time this test was written.
```

Comment:
I think the comment in `Digest::hash` is obsolete and can be removed. It's really just the blake2b hash now and there's no reason to use `blake2b_hash` explicitly anymore. (Please correct me if I'm wrong @fizyk20!)

Reply:
I believe I remove the commented code from there in my PR to `feat-fast-sync` - it should be okay to remove it here as well.

Reply:
We could also keep it, as it would be removed by the merge anyway. I don't want to extend the scope of this PR.